patroni/postgresql/config.py

```diff
@@ -1154,16 +1156,27 @@ def reload_config(self, config: Dict[str, Any], sighup: bool = False) -> None:
             if self._postgresql.major_version >= 90500:
                 time.sleep(1)
                 try:
-                    pending_restart = self._postgresql.query(
-                        'SELECT COUNT(*) FROM pg_catalog.pg_settings'
-                        ' WHERE pg_catalog.lower(name) != ALL(%s) AND pending_restart',
-                        [n.lower() for n in self._RECOVERY_PARAMETERS])[0][0] > 0
-                    self._postgresql.set_pending_restart(pending_restart)
+                    settings_diff = {
+                        p: (v, (lambda x: '?' if x is None else x)(self._postgresql.get_guc_value(p)))
+                        for p, v in self._postgresql.query(
+                            'SELECT name, setting FROM pg_catalog.pg_settings'
+                            ' WHERE pg_catalog.lower(name) != ALL(%s) AND pending_restart',
+                            [n.lower() for n in self._RECOVERY_PARAMETERS])
+                    }
+                    pending_restart = len(settings_diff) > 0
+                    external_change = settings_diff.keys() - param_diff.keys()
+                    if external_change:
+                        logger.info("PostgreSQL configuration parameters requiring restart"
+                                    " (%s) seem to be changed bypassing Patroni config."
+                                    " Setting 'Pending restart' flag", ', '.join(external_change))
```

Comment:
What about the case when parameter is set in Patroni config and in some externally managed file with a different value?